### ISSUE TITLE
Intermittent monitorKatoTask(Deploy) timeouts

### ISSUE THREAD

**Issue by peppe77** [spinnaker/spinnaker#6410]
K8S 1.18
Spinnaker: 1.23.5

Triggered concurrent deployments (via webhook) towards the same K8S cluster/namespace. We have been experiencing intermittent monitorKatoTask(deploy) timeouts. It can not be reproduced at will though happens rather often. Checked orca and clouddriver logs though nothing that hinted what is causing this problem. Let us know what additional logs/info are needed. thanks

```id":"01F3TRZ0C6X1JPKA5VGC37G8GK","refId":"2","type":"deployManifest","name":"Deploy (Manifest)","startTime":1619027528117,"endTime":1619031133919,"status":"TERMINAL","context":{"exception":{"exceptionType":"TimeoutException","shouldRetry":false,"details":{"stackTrace":"com.netflix.spinnaker.orca.exceptions.TimeoutException: MonitorKatoTask of stage Deploy (Manifest) timed out after 1 hour 0 minutes 4 seconds. pausedDuration: 0 seconds, elapsedTime: 1 hour 0 minutes 4 seconds, timeoutValue: 1 hour\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.checkForTaskTimeout(RunTaskHandler.kt:339)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.checkForTimeout(RunTaskHandler.kt:315)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.access$checkForTimeout(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$1$1.invoke(RunTaskHandler.kt:124)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$1$1.invoke(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withLoggingContext(RunTaskHandler.kt:416)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.access$withLoggingContext(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$1.invoke(RunTaskHandler.kt:105)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$1.invoke(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.AuthenticationAware$sam$java_util_concurrent_Callable$0.call(AuthenticationAware.kt)\n\tat com.netflix.spinnaker.security.AuthenticatedRequest.lambda$wrapCallableForPrincipal$0(AuthenticatedRequest.java:272)\n\tat com.netflix.spinnaker.orca.q.handler.AuthenticationAware$DefaultImpls.withAuth(AuthenticationAware.kt:53)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withAuth(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1.invoke(RunTaskHandler.kt:104)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1.invoke(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$withTask$1.invoke(RunTaskHandler.kt:244)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler$withTask$1.invoke(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withTask$1.invoke(OrcaMessageHandler.kt:68)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withTask$1.invoke(OrcaMessageHandler.kt:46)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withStage$1.invoke(OrcaMessageHandler.kt:85)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withStage$1.invoke(OrcaMessageHandler.kt:46)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withExecution(OrcaMessageHandler.kt:95)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withExecution(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withStage(OrcaMessageHandler.kt:74)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withStage(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withTask(OrcaMessageHandler.kt:60)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withTask(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withTask(RunTaskHandler.kt:233)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.handle(RunTaskHandler.kt:101)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.handle(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.q.MessageHandler$DefaultImpls.invoke(MessageHandler.kt:36)\n\tat com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.invoke(OrcaMessageHandler.kt)\n\tat com.netflix.spinnaker.orca.q.handler.RunTaskHandler.invoke(RunTaskHandler.kt:75)\n\tat com.netflix.spinnaker.orca.q.audit.ExecutionTrackingMessageHandlerPostProcessor$ExecutionTrackingMessageHandlerProxy.invoke(ExecutionTrackingMessageHandlerPostProcessor.kt:72)\n\tat com.netflix.spinnaker.q.QueueProcessor$callback$1$1.run(QueueProcessor.kt:89)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\n","error":"Unexpected Task Failure","errors":["MonitorKatoTask of stage Deploy (Manifest) timed out after 1 hour 0 minutes 4 seconds. pausedDuration: 0 seconds, elapsedTime: 1 hour 0 minutes 4 seconds, timeoutValue: 1 hour"]},"operation":"monitorDeploy","timestamp":1619031133903},"deploy.account.name":"k8s-ccdp-euwest1-dev-gen2","kato.task.terminalRetryCount":0,"moniker":{"app":"cdls-cart-service"},"source":"artifact","kato.task.firstNotFoundRetry":-1,"manifestArtifactId":"f95c4b26-632e-4315-a9a6-e37552419f3d","optionalArtifacts":```

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by peppe77** [comment]
@spinnakerbot remove-label to-be-closed

**Comment by abdulbasitkay** [comment]
Also have the same issue on spinnaker 1.26.4 with MySQL setup for clouddriver.

**Comment by sethev** [comment]
We see this too when deploying to k8s. Anything that caused the deploy task to stop running without successfully updating the task status would cause this.

I recently found a specific example, though, where the code was parked on this future: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-core/src/main/java/com/netflix/spinnaker/clouddriver/jobs/local/JobExecutorLocal.java#L114

The kubectl command had completed but i'm guessing there was a deadlock in the stream processing code.

Here's the stack trace I captured:
```
"DefaultOrchestrationProcessor-3" #227 prio=5 os_prio=0 cpu=3321.41ms elapsed=6805.00s tid=0x0000564d08b18800 nid=0x329 waiting on condition [0x00007fdce0d8e000]
java.lang.Thread.State: WAITING (parking)
at jdk.internal.misc.Unsafe.park(java.base@11.0.9/Native Method)
- parking to wait for <0x00000004627b5388> (a java.util.concurrent.FutureTask)
at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9/LockSupport.java:194)
at java.util.concurrent.FutureTask.awaitDone(java.base@11.0.9/FutureTask.java:447)
at java.util.concurrent.FutureTask.get(java.base@11.0.9/FutureTask.java:190)
at com.netflix.spinnaker.clouddriver.jobs.local.JobExecutorLocal.executeStreaming(JobExecutorLocal.java:114)
at com.netflix.spinnaker.clouddriver.jobs.local.JobExecutorLocal.lambda$runJob$0(JobExecutorLocal.java:56)
at com.netflix.spinnaker.clouddriver.jobs.local.JobExecutorLocal$$Lambda$1020/0x0000000800fb6c40.execute(Unknown Source)
at com.netflix.spinnaker.clouddriver.jobs.local.JobExecutorLocal.executeWrapper(JobExecutorLocal.java:66)
at com.netflix.spinnaker.clouddriver.jobs.local.JobExecutorLocal.runJob(JobExecutorLocal.java:56)
at com.netflix.spinnaker.clouddriver.kubernetes.op.job.KubectlJobExecutor.list(KubectlJobExecutor.java:407)
at com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesCredentials.lambda$list$7(KubernetesCredentials.java:469)
at com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesCredentials$$Lambda$1303/0x00000008012b8440.get(Unknown Source)
at com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesCredentials.runAndRecordMetrics(KubernetesCredentials.java:674)
at com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesCredentials.runAndRecordMetrics(KubernetesCredentials.java:659)
at com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesCredentials.list(KubernetesCredentials.java:465)
at com.netflix.spinnaker.clouddriver.kubernetes.caching.view.provider.ArtifactProvider.getArtifacts(ArtifactProvider.java:65)
at com.netflix.spinnaker.clouddriver.kubernetes.caching.view.provider.ArtifactProvider.getArtifacts(ArtifactProvider.java:56)
at com.netflix.spinnaker.clouddriver.kubernetes.op.artifact.KubernetesCleanupArtifactsOperation.artifactsToDelete(KubernetesCleanupArtifactsOperation.java:120)
at com.netflix.spinnaker.clouddriver.kubernetes.op.artifact.KubernetesCleanupArtifactsOperation$$Lambda$1338/0x00000008011cd440.apply(Unknown Source)
at java.util.stream.ReferencePipeline$3$1.accept(java.base@11.0.9/ReferencePipeline.java:195)
at java.util.HashMap$KeySpliterator.forEachRemaining(java.base@11.0.9/HashMap.java:1603)
at java.util.stream.AbstractPipeline.copyInto(java.base@11.0.9/AbstractPipeline.java:484)
at java.util.stream.AbstractPipeline.wrapAndCopyInto(java.base@11.0.9/AbstractPipeline.java:474)
at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(java.base@11.0.9/ReduceOps.java:913)
at java.util.stream.AbstractPipeline.evaluate(java.base@11.0.9/AbstractPipeline.java:234)
at java.util.stream.ReferencePipeline.collect(java.base@11.0.9/ReferencePipeline.java:578)
at com.netflix.spinnaker.clouddriver.kubernetes.op.artifact.KubernetesCleanupArtifactsOperation.operate(KubernetesCleanupArtifactsOperation.java:75)
at com.netflix.spinnaker.clouddriver.kubernetes.op.artifact.KubernetesCleanupArtifactsOperation.operate(KubernetesCleanupArtifactsOperation.java:46)
at com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation$operate.call(Unknown Source)
at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1$_closure2.doCall(DefaultOrchestrationProcessor.groovy:127)
at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1$_closure2.doCall(DefaultOrchestrationProcessor.groovy)
at jdk.internal.reflect.GeneratedMethodAccessor481.invoke(Unknown Source)
at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.9/DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(java.base@11.0.9/Method.java:566)
at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:101)
at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)
at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041)
at groovy.lang.Closure.call(Closure.java:405)
at groovy.lang.Closure.call(Closure.java:399)
at com.netflix.spinnaker.clouddriver.metrics.TimedCallable$CallableWrapper.call(TimedCallable.java:81)
at com.netflix.spinnaker.clouddriver.metrics.TimedCallable.call(TimedCallable.java:45)
at java_util_concurrent_Callable$call.call(Unknown Source)
at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1.doCall(DefaultOrchestrationProcessor.groovy:126)
at com.netflix.spinnaker.clouddriver.orchestration.DefaultOrchestrationProcessor$_process_closure1.doCall(DefaultOrchestrationProcessor.groovy)
at jdk.internal.reflect.GeneratedMethodAccessor476.invoke(Unknown Source)
at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.9/DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(java.base@11.0.9/Method.java:566)
at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:101)
at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)
at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041)
at groovy.lang.Closure.call(Closure.java:405)
at groovy.lang.Closure.call(Closure.java:399)
at com.netflix.spinnaker.security.AuthenticatedRequest.lambda$wrapCallableForPrincipal$0(AuthenticatedRequest.java:272)
at com.netflix.spinnaker.security.AuthenticatedRequest$$Lambda$1272/0x00000008011f0840.call(Unknown Source)
at com.netflix.spinnaker.clouddriver.metrics.TimedCallable.call(TimedCallable.java:45)
at java.util.concurrent.FutureTask.run(java.base@11.0.9/FutureTask.java:264)
at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9/ThreadPoolExecutor.java:1128)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9/ThreadPoolExecutor.java:628)
at java.lang.Thread.run(java.base@11.0.9/Thread.java:834)
```

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by spinnakerbot** [comment]
This issue is tagged as 'to-be-closed' and hasn't been updated  in 45 days, so we are closing it. You can always reopen this issue if needed.
